avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -619,82 +676,121 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
-    // struct converts to one of those (record), so for everything else that has a name we store
-    // the full name into a special property. For uniformity, we also duplicate this info into
-    // the same field in records as well even though it will also be available in the namespace()
-    // and name().
-    if (schema.name() != null) {
-      baseSchema.addProp(CONNECT_NAME_PROP, schema.name());
-
-      if(Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())){
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DECIMAL);
-      } else if(Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIME_MILLIS);
-      } else if(Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIMESTAMP_MILLIS);
-      } else if(Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
-        baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DATE);
+        if (schema.name() != null) {
+          baseSchema.addProp(CONNECT_NAME_PROP, schema.name());
+        }
+        // Some Connect types need special annotations to preserve the types accurate due to
+        // limitations in Avro. These types get an extra annotation with their Connect type
+        if (connectType != null) {
+          baseSchema.addProp(CONNECT_TYPE_PROP, connectType);
+        }
+      }
+      
+      // Only Avro named types (record, enum, fixed) may contain namespace + name. Only Connect's
+      // struct converts to one of those (record), so for everything else that has a name we store
+      // the full name into a special property. For uniformity, we also duplicate this info into
+      // the same field in records as well even though it will also be available in the namespace()
+      // and name().
+      if (schema.name() != null) {
+        if(Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())){
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DECIMAL);
+        } else if(Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIME_MILLIS);
+        } else if(Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_TIMESTAMP_MILLIS);
+        } else if(Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          baseSchema.addProp(AVRO_LOGICAL_TYPE_PROP, AVRO_LOGICAL_DATE);
+        }
       }
-    }
 
-    // Some Connect types need special annotations to preserve the types accurate due to
-    // limitations in Avro. These types get an extra annotation with their Connect type
-    if (connectType != null) {
-      baseSchema.addProp(CONNECT_TYPE_PROP, connectType);
-    }
+      if (schema.parameters() != null){
+        for(Map.Entry<String, String> entry : schema.parameters().entrySet())
+        {
+          if (entry.getKey().startsWith(AVRO_PROP)){
```

Comment:
I was actually more concerned that it's too narrow a filter rather than too broad. I figured that if a connector generates a schema with other parameters that aren't prefixed with `avro`, we won't currently attempt to preserve them. Even if we didn't do anything with them, Avro provides enough support to pass them through the schema such that they could be picked up by another downstream connector that understood how to handle them.